File: src/Implicit/ImplicitConstant.java
package implicit;

public final class ImplicitConstant extends ImplicitModuleBase {
    private final double value;
    
    public ImplicitConstant() {
        this.value = 0.00;
    }
    
    public ImplicitConstant(final double value) {
        this.value = value;
    }
    
    @Override
    public double get(final double x, final double y) {
        return this.value;
    }
    
    @Override
    public double get(final double x, final double y, final double z) {
        return this.value;
    }
    
    @Override
    public double get(final double x, final double y, final double z, final double w) {
        return this.value;
    }
    
    @Override
    public double get(final double x, final double y, final double z, final double w, final double u, final double v) {
        return this.value;
    }
}

File: src/Implicit/ImplicitConstant.cs
using System;

namespace TinkerWorX.AccidentalNoiseLibrary
{
    public sealed class ImplicitConstant : ImplicitModuleBase
    {
        public ImplicitConstant()
        {
            this.Value = 0.00;
        }

        public ImplicitConstant(double value)
        {
            this.Value = value;
        }

        public double Value { get;  set; }

        public override double Get(double x, double y)
        {
            return this.Value;
        }

        public override double Get(double x, double y, double z)
        {
            return this.Value;
        }

        public override double Get(double x, double y, double z, double w)
        {
            return this.Value;
        }

        public override double Get(double x, double y, double z, double w, double u, double v)
        {
            return this.Value;
        }
    }
}
